File: Selenium/Activities/Activity14.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.interactions.Action;
import org.openqa.selenium.Keys;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class Activity14 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver wd = new FirefoxDriver();
        wd.manage().window().fullscreen();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(wd, Duration.ofSeconds(10));

        wd.get("https://v1.training-support.net/selenium/tables");
        System.out.println("Title of the home page is -> : " + wd.getTitle());

        List<WebElement> rows = wd.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        System.out.println("Number of rows in the table are -> " + rows.size());

        List<WebElement> cols = wd.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("Number of columns in the table are -> " + cols.size());

        WebElement cellValue = wd.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println("Second row, second cell value: " + cellValue.getText());

        wd.findElement(By.xpath("//table[@id=\"sortableTable\"]/thead/tr/th")).click();

        cellValue = wd.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));

        System.out.println("Second row, second cell value: " + cellValue.getText());

        List<WebElement> footer = wd.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        System.out.println("Cell values of the footer: ");
        for (WebElement cell : footer) {
            System.out.println(cell.getText());
            System.out.println(cell.getText());

        }
        wd.quit();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("https://v1.training-support.net/selenium/tables")
    #driver.set_window_size(1024, 600)
    driver.maximize_window()
    # Print the title of the page
    print("Page title is: ", driver.title)

    # Get number of columns
    columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/thead/tr/th")
    print("Number of columns: ", len(columns))
    # Get number of rows
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    print("Number of rows: ", len(rows))

    # Get cell value from the second row and second column before sorting
    second_row_second_cell_before_sort = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    # Print the cell value
    print("Cell value at second row and second column: ", second_row_second_cell_before_sort.text)

    # Find and click the first heading in the table
    driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]").click()

    # Get cell value from the second row and second column after sorting
    second_row_second_cell_after_sort = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    # Print the cell value
    print("Cell value at second row and second column: ", second_row_second_cell_after_sort.text)